internal/remoteconfig/remoteconfig.go

```diff
@@ -92,37 +102,65 @@ func NewClient(config ClientConfig) (*Client, error) {
 		stop:         make(chan struct{}),
 		lastError:    nil,
 		callbacks:    []Callback{},
+		capabilities: map[Capability]struct{}{},
+		products:     map[string]struct{}{},
 	}, nil
 }
 
-// Start starts the client's update poll loop in a fresh goroutine
-func (c *Client) Start() {
-	go func() {
-		ticker := time.NewTicker(c.PollInterval)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-c.stop:
-				close(c.stop)
-				return
-			case <-ticker.C:
-				c.updateState()
+// Start starts the client's update poll loop in a fresh goroutine.
+// Noop if the client has already started.
+func Start(config ClientConfig) error {
+	if client != nil {
```

Comment:
I don't see the client being set to nil in `Stop()`. This means that the global client can't be restarted after a call to `Stop()` unless explicitly calling `Reset()`. Is this intended? We should probably document it if that's the case.

Reply:
Correct! Added it in the docstring - the mean reason of this behavior is to limit the need to call `Reset()` in the tests.